File: 위장/seonjuleee/위장.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        for (int i=0; i<clothes.length; i++) {
            if (map.containsKey(clothes[i][1])) {
                map.replace(clothes[i][1],map.get(clothes[i][1]) + 1);
            } else {
                map.put(clothes[i][1], 1);
            }
        }
        for (String k : map.keySet()) {
            answer *= map.get(k) + 1;
        }
        return answer - 1;
    }
}

File: 위장/tjdyyy/위장.py
def solution(clothes):
    answer = 1
    
    dic = {} # 딕셔너리 사용
    
    for list in clothes:
        if list[1] in dic: # 딕셔너리에 옷 종류가 있으면
            dic[list[1]] += 1 # 값 증가
        else: # 없으면
            dic[list[1]] = 1 # 딕셔너리 생성
        
    for i in dic.values():
        answer *= (i+1) # 종류 별 개수 + 1씩 곱함
        
    return answer-1 # 아무것도 안 입는 경우 -1을 뺌
